### ISSUE TITLE
Certain recipes fail to remove RAW penalty from food

### ISSUE THREAD

**Issue by Salty-Panda** [CleverRaven/Cataclysm-DDA#48569]
# Describe the bug

In code, there is a function `hot_result()` that checks whether a recipe involves the usage of a hotplate. If so, it adds COOKED flag. When it happens, resulting items lose RAW penalty.

Some recipes should remove that flag too, according to data of result items, but fail to do so.
E.g.:
Both salted fish and pickled fish lack the RAW flag.
The crafted pickled fish has the same calorie value as the raw fish though.
In the case of salted fish, there are two recipes available, one makes use of FIRE, the other not. Only the former properly increases the calorie value of the fish.

Tested using game report, calorie gained also is too low, not just display.

# Steps To Reproduce

Check recipes for salted fish, only one uses hotplate/cooking tool.
Use both of them, compare values.

# Expected behavior

Crafting checks whether the resulting item has RAW flag. It could possibly cause some minor problems if there are improperly marked items.
EDIT. Edited `hot_result()`. It's a bad idea, items will be also heated and recipes will allow using frozen components.

Or add some flag or recipe property, that would also allow removing the RAW flag.

Possibly can be also fixed at lower level by properly propagating flags in `compute_effective_nutrients` callstack.

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 2004
- Game Version: 0.E-10383-gad91f6a [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Aftershock [aftershock],
    Blaze Industries [blazeindustries],
    Magiclysm [magiclysm],
    Graphical Overmap [Graphical_Overmap],
    Graphical Overmap Magiclysm [Graphical_Overmap_Magiclysm],
    Mythical Martial Arts [MMA],
    C.R.I.T Expansion Mod [crt_expansion],
    No Fungal Monsters [No_Fungi],
    Stats Through Kills [stats_through_kills],
    SpeedyDex [speedydex],
    Bionic Professions [package_bionic_professions],
    Aftershock - UndeadPeople Tileset [AftershockUDP],
    Blaze Industries - UndeadPeople Tileset [BlazeIndustriesUDP],
    C.R.I.T Expansion Mod - UndeadPeople Tileset [crt_expansionUDP],
    Magiclysm - UndeadPeople Tileset [MagiclysmUDP],
    Mythical Martial Arts - UndeadPeople Tileset [MMAUDP]
]

**Comment by actual-nh** [comment]
There are also some oddball cases I noticed but have never gotten around to checking further into - like beef jerky having the RAW flag.

**Comment by Anaglyphic** [comment]
Why closed? It's still this way.

Since the title is "Certain recipes" and not a specific one, I'll add reference to actual-nh's comment, These particular ones seems an error, but I wonder if most of these recipe issues (like the allergen stuff) highlight that lack of granularity issue, like what I-am-Erk brings up repeatedly in various subsystems needing overhaul...

It is a very "_huh?"_ moment when your smoked-in-a-charcoal-smoker jerky, made from pre-cooked meat (the interstage from a hunk o meat) pops out as "meat jerky (hot)" and yet still retains the raw flag.

![smoked-jerky-made-from-cooked-meat-is-still-considered-raw](https://user-images.githubusercontent.com/19985100/117677866-d60a6a80-b17c-11eb-8dff-137838fbc392.gif)

**Comment by Salty-Panda** [comment]
The jerky you linked has 402 kcals and not 300 which means it has got a COOKED flag.
I suggest making separate issue, since it might be a bug with smoking rack.

**Comment by Anaglyphic** [comment]
Yea I'm sure it's me. Sorry. I'm a bit unorganized in my git builds. I'll shut up and figure out what dumb thing I did, I do see ex-16 has a commit #48570 for this, _recipes_properly_remove_raw_penalty_, checked in by you, in fact! So somewhere between ex-6 and here I messed up my build trees... I'm building ex-18 (#define VERSION "0.E-10519-g272a1d4244") right now, I'm sure it'll be fixed.

Disregard my constant stupidity.
